examples/common/chip-app-server/QRCodeUtil.cpp

```diff
@@ -75,3 +88,44 @@ CHIP_ERROR GetQRCode(uint32_t & setupPinCode, std::string & QRCode, chip::Rendez
 exit:
     return err;
 }
+
+static inline bool isCharUnreservedInRfc3986(const char c)
+{
+    return isalpha(c) || isdigit(c) || (strchr(specialCharsUnreservedInRfc3986, c) != nullptr);
+}
+
+CHIP_ERROR EncodeQRCodeToUrl(const char * QRCode, size_t len, char * url, size_t maxSize)
+{
+    const char upperHexDigits[] = "0123456789ABCDEF";
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    size_t i = 0, j = 0;
+
+    VerifyOrExit((QRCode && url) != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Comparing a boolean to NULL is pretty weird. This should probably be `(QRCode != nullptr) && (url != nullptr)`